lib/trackler/problem.rb

```diff
@@ -10,78 +10,99 @@ def initialize(slug, root)
     end
 
     def exists?
-      !!md && !!yaml
+      !!description && !!metadata
     end
 
     def name
-      @name ||= slug.split('-').map(&:capitalize).join(' ')
+      slug.split('-').map(&:capitalize).join(' ')
     end
 
     def description
-      @description ||= File.read(filepath(md))
+      @description ||= if File.exists?(common_metadata_path(description_path))
+                         File.read(common_metadata_path(description_path))
+                       end
     end
 
     def source_markdown
-      body = source.empty? ? "" : "%s" % source
-      url = source_url.empty? ? "" : "[%s](%s)" % [source_url, source_url]
-      if url.empty? && body.empty?
-        ""
-      else
-        "## Source\n\n" + [body, url].reject(&:empty?).join(" ")
-      end
+      text = [source, markdown_link(source_url)].reject(&:empty?).join(" ")
+      text.empty? ? text : "## Source\n\n#{text}"
     end
 
+    ######
+    # Deprecated methods
+    # TODO: remove external references to these methods.
+    # found in: x-api
+    # NOT in: exercism.io, cli
+    # Anywhere else we need to look?
+    # Should this output a warning or raise an error?
```

Comment:
Let's not raise a warning or error—let's just quietly deprecate them, update x-api, and then remove them.